File: Data-Structure/Java/LeetCode/Array/1428. Leftmost Column with at Least a One.java
//1428. Leftmost Column with at Least a One
//https://leetcode.com/problems/leftmost-column-with-at-least-a-one/
/*
Input: mat = [[0,0,0,1],[0,0,1,1],[0,1,1,1]]
Output: 1
*/
class Solution {
    public int leftMostColumnWithOne(BinaryMatrix binaryMatrix) {
        int rows = binaryMatrix.dimensions().get(0);
        int cols = binaryMatrix.dimensions().get(1);

        // Set pointers to the top-right corner.
        int currRow = 0;
        int currCol = cols - 1;

        // Repeat the search until it goes off the grid.
        while (currRow < rows && currCol >= 0) {
            if (binaryMatrix.get(currRow, currCol) == 0) {
                currRow++;
            } else {
                currCol--;
            }
        }

        // If we never left the last column, this is because it was all 0's.
        return (currCol == cols - 1) ? -1 : currCol + 1;
    }
}
// Shortest Word Distance
// Number of Subarrays with Bounded Maximum
// Average Waiting Time

File: Data-Structure/Nodejs/LeetCode All in one/1428. Leftmost Column with at Least a One.js
//1428. Leftmost Column with at Least a One
{
    var leftMostColumnWithOne = function (binaryMatrix) {
        const [rows, cols] = binaryMatrix.dimensions();

        // Set pointers to the top-right corner.
        let currentRow = 0;
        let currentCol = cols - 1;

        // Repeat the search until it goes off the grid.
        while (currentRow < rows && currentCol >= 0) {
            if (binaryMatrix.get(currentRow, currentCol) === 0) {
                currentRow++;
            } else {
                currentCol--;
            }
        }

        // If we never left the last column, this is because it was all 0's.
        return (currentCol == cols - 1) ? -1 : currentCol + 1;
    };
    //Shortest Word Distance
    //Number of Subarrays with Bounded Maximum
    //Average Waiting Time
}